website/docs: document how to use Google's SMTP relay (#15040)
goauthentik/authentik: website/docs/troubleshooting/emails.md

```shell
kubectl exec -it deployment/authentik-worker -c worker -- ak test_email [...]
```

## Google Workspace SMTP relay configuration

> To send email through Google SMTP servers, it is often easiest to use [Google's Relay option](https://support.google.com/a/answer/2956491). Google has documentation available at [Send email from a printer, scanner, or app](https://support.google.com/a/answer/176600?hl=en). Setting this up can be tricky as there is conflicting information about correct settings online. While setting up a service account and an App Password is possible, it is not needed to successfully send emails from authentik.

Reviewer: ```suggestion
To send email through Google SMTP servers, the easiest and most reliable method is often to use [Google's SMTP relay service](https://support.google.com/a/answer/2956491). Google provides detailed guidance in their documentation: [Send email from a printer, scanner, or app](https://support.google.com/a/answer/176600?hl=en). Configuration can be confusing, as there’s a lot of conflicting information available online. While it is technically possible to use a service account or an App Password, these are not required to successfully send emails from authentik.
```

Reviewer: Also, I'm probably skipping a big chunk of information, but are we talking about a Google or authentik SA / App password? Either way, using one would be the recommended way of doing it in my book and should be privileged.

Reviewer: I'd also remove the sentence of it being confusing, as we provide all required information here and don't want to set a sort of ambiguity if you get what I mean

Reviewer: Sorry for the delay, LGTM

> To send email through Google SMTP servers, it is often easiest to use [Google's Relay option](https://support.google.com/a/answer/2956491). For Authentik, Google has easier to follow documentation available at [Send email from a printer, scanner, or app](https://support.google.com/a/answer/176600?hl=en). Setting this up can be tricky as there is conflicting information about correct settings online. While setting up a service account and an App Password is possible, it is not needed to successfully send emails from Authentik.

Reviewer 2: ```suggestion
To send email through Google SMTP servers, it is often easiest to use [Google's Relay option](https://support.google.com/a/answer/2956491). Google has documentation available at [Send email from a printer, scanner, or app](https://support.google.com/a/answer/176600?hl=en). Setting this up can be tricky as there is conflicting information about correct settings online. While setting up a service account and an App Password is possible, it is not needed to successfully send emails from authentik.
```

> First, confirm the outbound IP address that authentik will use to send emails with. Follow Google's documentation to add the IP address or addresses to the "SMTP relay service" options in your workspace's Gmail settings.

Reviewer: If it's not too complicated, it would be preferred to include a few steps on how to do this / link official docs

Author: The docs are linked to in the previous paragraph. I'm hesitant to copy them here given they could change. Do you want them linked to again?

> First, confirm the outbound IP address that Authentik will use to send emails with. Follow Google's documentation to add the IP address or addresses to the "SMTP relay service" options in your workspace's Gmail settings.

Reviewer 2: ```suggestion
First, confirm the outbound IP address that authentik will use to send emails with. Follow Google's documentation to add the IP address or addresses to the "SMTP relay service" options in your workspace's Gmail settings.
```

Reviewer 2: Can we tell them where/how to confirm that the outbound IP address is, that authentik will use?

Author: There's a few ways, ranging from manually inspecting things with `ip`, or `curl`'ing ifconfig.co, icanhazip.com, etc. Any preference?

- Set Allowed Senders to `Only addresses in my domains`.

> - Set Authentication to `Only accept mail from the specified IP addresses`.

Reviewer 2: ```suggestion
- Set **Authentication** to `Only accept mail from the specified IP addresses`.
```

> - Do not set `Require SMTP Authentication`.

Reviewer 2: ```suggestion
- Do not set **Require SMTP Authentication**.
```

> - Set `Require TLS encryption`.

Reviewer 2: ```suggestion
- Select **Require TLS encryption**.
```

> Then, set the following environment variables for authentik:

Reviewer: The setup would be different for k8s and I'd assume a fair part of Google Workspace setups (which is required for this bit) would use an enterprise solution like k8s

Author: I'm not that familiar with k8s, but wouldn't you still need these environment variables? Or are they named differently?

Author: I took a pass at this. I'm not sure if the convention is to omit optional unused items, or to leave them in as empty strings.

> Then, set the following environment variables for Authentik:

Reviewer 2: ```suggestion
Then, set the following environment variables for authentik:
```

```
AUTHENTIK_EMAIL__HOST=smtp-relay.gmail.com
AUTHENTIK_EMAIL__PORT=587
AUTHENTIK_EMAIL__USE_TLS=true
AUTHENTIK_EMAIL__USE_SSL=false
AUTHENTIK_EMAIL__TIMEOUT=10
```

> Recreate the Authentik containers, and use the `ak test_email` command to confirm email is working.

Reviewer 2: ```suggestion
Recreate the authentik containers, and use the `ak test_email` command to confirm email is working.
```
